From 6b86192acb3b1c33a6d4701ca8aea718fc98c84b Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 10 Mar 2020 16:39:37 +0100
Subject: [PATCH] PONRTSYS-6861: Allow PON FW to access CGU register

It looks like the PON IP FW access the CGU over the CPU3 bits.
This is needed to make the PON loop timing work.

I was unable to access this register from the InterAptiv CPU, but
a full test needs better understanding of the NGI firewall.
---
 arch/mips/lantiq/grx500_bootcore/ngi_prx300.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
index 191ea292b3d0..070608439a19 100644
--- a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
+++ b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
@@ -277,9 +277,9 @@ void set_permission(void)
 	permission_wr(ssx4_base,
 			REG4_TREG4_PM_TREG4_PM_READ_PERMISSION_0, M4KEC_ONLY);
 	permission_wr(ssx4_base,
-			REG4_TCGU_PM_TCGU_PM_WRITE_PERMISSION_0, M4KEC_ONLY);
+			REG4_TCGU_PM_TCGU_PM_WRITE_PERMISSION_0, M4KEC_ONLY | CPU3);
 	permission_wr(ssx4_base,
-			REG4_TCGU_PM_TCGU_PM_READ_PERMISSION_0, M4KEC_ONLY);
+			REG4_TCGU_PM_TCGU_PM_READ_PERMISSION_0, M4KEC_ONLY | CPU3);
 	permission_wr(ssx4_base,
 			REG4_TRCU_PM_TRCU_PM_WRITE_PERMISSION_0, M4KEC_ONLY);
 	permission_wr(ssx4_base,
